src/pages/MainPage.vue

```diff
@@ -295,6 +295,8 @@ export default {
           _this.$store.commit('setRetrieved', newRoad);
         });
       } else if (newRoad !== '') {
+        // Reset fulfillmentNeeded so that new Roads still call FireRoad to check fulfillment.
+        this.$store.commit('resetFulfillmentNeeded');
```

Comment:
This is calling update fulfillment twice when you create a new road while logged in (once on creation, once on name change).  This should go specifically after a new road is added instead (in addRoad on MainPage).